src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -5667,5 +5862,106 @@ void getParts(BoundInterpolatedString interpolatedString)
                 }
             }
         }
+
+        private SafeContext GetValEscapeOfInterpolatedStringHandlerCalls(BoundExpression expression, SafeContext localScopeDepth)
+        {
+            SafeContext scope = SafeContext.CallingMethod;
+
+            while (true)
+            {
+                switch (expression)
+                {
+                    case BoundBinaryOperator binary:
+                        scope = scope.Intersect(GetValEscapeOfInterpolatedStringHandlerCalls(binary.Right, localScopeDepth));
+                        expression = binary.Left;
+                        break;
+
+                    case BoundInterpolatedString interpolatedString:
+                        return scope.Intersect(getParts(interpolatedString, localScopeDepth));
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(expression.Kind);
+                }
+            }
+
+            SafeContext getParts(BoundInterpolatedString interpolatedString, SafeContext localScopeDepth)
+            {
+                SafeContext scope = SafeContext.CallingMethod;
+
+                foreach (var part in interpolatedString.Parts)
+                {
+                    if (part is not BoundCall { Method.Name: BoundInterpolatedString.AppendFormattedMethod } call)
+                    {
+                        // Dynamic calls cannot have ref struct parameters, and AppendLiteral calls will always have literal
+                        // string arguments and do not require us to be concerned with escape
```

Comment:
>and do not require us to be concerned with escape [](http://example.com/codeflow?start=44&length=49)

Can ```AppendLiteral``` have an `in` parameter? I assume the ref-ness of the parameter is important, not just the argument value.

Reply:
Yes, AppendLiteral calls might have the same issue, thanks.